src/xcape/core.py

```diff
@@ -14,9 +15,17 @@
 from .stdheight import stdheight as _stdheight
 
 def _prod(v):
+    '''
+    Lambda function for products between .
+    '''
     return reduce(lambda x, y: x*y, v)
 
 def _reshape_inputs(*args):
+    '''
+    Function to reshape multiple arrays of input data from native 4D (X,Y,T,Z) format to a 2D 
+    vertical array (XYT,Z) to allow for parallelization of calculation of desired parameter. 
+    Called by functions _calc_*_numpy. 
+    '''
```

Comment:
I think this is built to take any ND array, so it could be also (X,Y,T,M,Z) , say for M ensemble members, it just flattens what is NOT the vertical dimension together. 
you could change to : input data from, i.e., native 4D (X,Y,T,Z) format to a 2D vertical array (XYT,Z), or any (X1,X2,X3,X4..., Z) to (X1*X2*X3*X4...., Z)